### PR TITLE
Scientists react to gunshots

### PR DESCRIPTION
**Pull request by TheArtemMaps**
HL1 Scientist, Keller and Rosenberg react to gunshots

### CODE REVIEW

File: lua/entities/npc_vj_hlr1_scientist/init.lua [hunk 1]
```diff
@@ -312,6 +312,36 @@ function ENT:CustomOnThink()
...
+}
+
+hook.Add("EntityFireBullets", "BulletFiredHook", function(ent, data)
```
Comment (DrVrej) [inline lua/entities/npc_vj_hlr1_scientist/init.lua:321]: This is highly unoptimized with many fail cases

File: lua/entities/npc_vj_hlr1_scientist/init.lua [hunk 2]
```diff
@@ -338,6 +368,10 @@ end
 function ENT:CustomOnPriorToKilled(dmginfo, hitgroup)
 	self:SetBodygroup(2, 0)
 end
+
+function ENT:CustomOnRemove()
+hook.Remove( "EntityFireBullets", "BulletFiredHook" )
```
Comment (DrVrej) [inline lua/entities/npc_vj_hlr1_scientist/init.lua:373]: Wrong indentation and spacings

File: lua/entities/npc_vj_hlrbs_rosenberg/init.lua [hunk 3]
```diff
@@ -84,4 +84,41 @@ function ENT:SCI_CustomOnInitialize()
...
 
 	self:SetBodygroup(1, 5)
+end
+
+local lastSoundTime = 0
+local soundList = {
+"vj_hlr/hl1_npc/rosenberg/ro_plfear1.wav","vj_hlr/hl1_npc/rosenberg/ro_plfear2.wav","vj_hlr/hl1_npc/rosenberg/ro_plfear3.wav","vj_hlr/hl1_npc/rosenberg/ro_plfear4.wav","vj_hlr/hl1_npc/rosenberg/ro_fear0.wav","vj_hlr/hl1_npc/rosenberg/ro_fear1.wav","vj_hlr/hl1_npc/rosenberg/ro_fear2.wav","vj_hlr/hl1_npc/rosenberg/ro_fear3.wav","vj_hlr/hl1_npc/rosenberg/ro_fear4.wav","vj_hlr/hl1_npc/rosenberg/ro_fear5.wav"
+}
+
+hook.Add("EntityFireBullets", "BulletFiredHook3", function(ent, data)
```
Comment (DrVrej) [inline lua/entities/npc_vj_hlrbs_rosenberg/init.lua:94]: Same as above

File: lua/entities/npc_vj_hlrdc_keller/init.lua [hunk 4]
```diff
@@ -84,9 +84,43 @@ function ENT:CustomOnDeath_BeforeCorpseSpawned(dmginfo, hitgroup)
 	self:SetBodygroup(0, 1)
 	self:SetPos(self:GetPos() + self:GetUp()*5)
 end
+
+local lastSoundTime = 0
+local soundList = {
+"vj_hlr/hl1_npc/keller/dk_help0.wav","vj_hlr/hl1_npc/keller/dk_help1.wav","vj_hlr/hl1_npc/keller/dk_help2.wav","vj_hlr/hl1_npc/keller/dk_fear0.wav","vj_hlr/hl1_npc/keller/dk_fear1.wav","vj_hlr/hl1_npc/keller/dk_fear2.wav","vj_hlr/hl1_npc/keller/dk_fear3.wav","vj_hlr/hl1_npc/keller/dk_fear4.wav","vj_hlr/hl1_npc/keller/dk_fear5.wav","vj_hlr/hl1_npc/keller/dk_fear6.wav"
+}
+
+hook.Add("EntityFireBullets", "BulletFiredHook2", function(ent, data)
```
Comment (DrVrej) [inline lua/entities/npc_vj_hlrdc_keller/init.lua:93]: Same as above

**Review by DrVrej — COMMENTED** [review]
Thank you for the contribution, but I have to kindly decline this. Mainly because the code is highly unoptimized and there are multiple fail cases with unhandled behaviors. There are much better and cleaner ways to implement this but I don't have such plans at the moment.

**Comment by TheArtemMaps** [comment]
> Thank you for the contribution, but I have to kindly decline this. Mainly because the code is highly unoptimized and there are multiple fail cases with unhandled behaviors. There are much better and cleaner ways to implement this but I don't have such plans at the moment.

No problem, i understand.

**Pull request closed without merging**